### PR TITLE
Add support for Optional.fromPath

### PR DESCRIPTION
**Pull request by cybermaak**
**Before submitting a pull request,** please make sure the following is done:

- Fork [the repository](https://github.com/gcanti/monocle-ts) and create your branch from `master`.
- Run `npm install` in the repository root.
- If you've fixed a bug or added code that should be tested, add tests!
- Ensure the test suite passes (`npm test`).

**Note**. If you find a typo in the **documentation**, make sure to modify the corresponding source (docs are generated).

### CODE REVIEW

**Comment by gcanti** [comment]
@cybermaak while reviewing your PR I found out that the current implementation of `Optional.fromNullableProp` is [buggy](https://github.com/gcanti/monocle-ts/issues/106)

File: src/index.ts [hunk 1]
```diff
@@ -719,6 +751,53 @@ export class Optional<S, A> {
...
+   *     employment: {}
+   *   }
+   * }
+   *
+   * numberFromResponse.getOption(response1) // some('555-1234')
+   * numberFromResponse.getOption(response2) // none
+   *
+   * @since 2.0.1
+   */
+  static fromNullablePath<S>(): OptionalFromPath<S> {
```
Comment (gcanti) [inline src/index.ts]: I'd rename this function to `fromPath` since it works even if all props are required

Comment (cybermaak) [reply src/index.ts]: Make sense. 
Renamed to `fromPath` and updated the PR title accordingly

**Comment by gcanti** [comment]
Thanks @cybermaak

**Pull request merged**